event_webhook_types.go

```diff
@@ -402,102 +402,127 @@ type MemberEvent struct {
+		Name      string `json:"name"`
+		Username  string `json:"username"`
+		AvatarURL string `json:"avatar_url"`
+		Email     string `json:"email"`
+	} `json:"user"`
+	ProjectID int `json:"project_id"`
+	Project   struct {
+		ID                int         `json:"id"`
+		Name              string      `json:"name"`
+		Description       string      `json:"description"`
+		WebURL            string      `json:"web_url"`
+		AvatarURL         interface{} `json:"avatar_url"`
+		GitSSHURL         string      `json:"git_ssh_url"`
+		GitHTTPURL        string      `json:"git_http_url"`
+		Namespace         string      `json:"namespace"`
+		VisibilityLevel   int         `json:"visibility_level"`
+		PathWithNamespace string      `json:"path_with_namespace"`
+		DefaultBranch     string      `json:"default_branch"`
+		Homepage          string      `json:"homepage"`
+		URL               string      `json:"url"`
+		SSHURL            string      `json:"ssh_url"`
+		HTTPURL           string      `json:"http_url"`
 	} `json:"project"`
 	ObjectAttributes struct {
-		Attachment       string        `json:"attachment"`
-		AuthorID         int           `json:"author_id"`
-		ChangePosition   *NotePosition `json:"change_position"`
-		CommitID         string        `json:"commit_id"`
-		CreatedAt        string        `json:"created_at"`
-		DiscussionID     string        `json:"discussion_id"`
-		ID               int           `json:"id"`
-		LineCode         string        `json:"line_code"`
-		Note             string        `json:"note"`
-		NoteableID       int           `json:"noteable_id"`
-		NoteableType     string        `json:"noteable_type"`
-		OriginalPosition *NotePosition `json:"original_position"`
-		Position         *NotePosition `json:"position"`
-		ProjectID        int           `json:"project_id"`
-		ResolvedAt       string        `json:"resolved_at"`
-		ResolvedByID     int           `json:"resolved_by_id"`
-		ResolvedByPush   bool          `json:"resolved_by_push"`
-		StDiff           *Diff         `json:"st_diff"`
-		System           bool          `json:"system"`
-		Type             string        `json:"type"`
-		UpdatedAt        string        `json:"updated_at"`
-		UpdatedByID      string        `json:"updated_by_id"`
-		Description      string        `json:"description"`
-		URL              string        `json:"url"`
+		ID           int         `json:"id"`
+		Note         string      `json:"note"`
+		NoteableType string      `json:"noteable_type"`
+		AuthorID     int         `json:"author_id"`
+		CreatedAt    string      `json:"created_at"`
+		UpdatedAt    string      `json:"updated_at"`
+		ProjectID    int         `json:"project_id"`
+		Attachment   interface{} `json:"attachment"`
+		LineCode     interface{} `json:"line_code"`
+		CommitID     string      `json:"commit_id"`
+		NoteableID   int         `json:"noteable_id"`
+		System       bool        `json:"system"`
+		StDiff       interface{} `json:"st_diff"`
+		URL          string      `json:"url"`
```

Comment:
Are you sure all the `ObjectAttributes` fields you removed are incorrect? I cannot imagine these fields ever being added without them being in the docs.

Reply:
Well, I'm not 100% sure about older version of Gitlab, but I just convert latest version of json event to Go struct and add it :-)  Before I did this I have got an error.  
Also this variant already in use and as far as I see, it works flawlessly. 

I think the struct in go-git was correct for previous version, let me check this, I will back with result a bit later.

Reply:
Yes, but we would like this package to be compatible with older versions of the GitLab API as well. in some cases we then add a deprecation notice (search the code for examples) to follow the GitLab API, but we try to not remove it to remain backwards compatible with older versions.